### ISSUE TITLE
Binary Code Signing on macOS

### ISSUE THREAD

**Issue by rcurranmoz** [taskcluster/taskcluster#7413]
We're encountering test failures as we transition to macOS 14 because the `worker-runner` binary is not signed.

Details: [https://bugzilla.mozilla.org/show_bug.cgi?id=1932140](https://bugzilla.mozilla.org/show_bug.cgi?id=1932140)

In summary, we can use the following command as a workaround: `codesign --force --sign - /usr/local/bin/start-worker`

However, this is likely not an ideal long-term solution.

Thank you

**Comment by bhearsum** [comment]
Any signing and notarization has to run through RelEng signing infra (that's the only place we have the signing key and notarization credentials available). The (_very_) short term workaround for this would be to sign and notarize the binary through https://github.com/mozilla-releng/adhoc-signing. This will be annoying to do on a regular basis, but we do use it to sign certain other things like mozregression.

 If we really need to sign and notarize I strongly suggest that we hook up Taskcluster releases to signingscript to automate this.

(Seeing as this is internal only, I think signing as you are is not the end of the world, although it is obviously also annoying!)

**Comment by rcurranmoz** [comment]
@petemoore 

Developer-ID signing the worker is the last blocker to zero-touch, EACS-able macOS CI provisioning

We're moving the Firefox-CI macOS test fleet (Apple Silicon M4, macOS 15, in mdc1) to SIP-enabled — and the driver is datacenter operations, not just security posture:

- SIP-off can't be provisioned zero-touch. Disabling SIP requires csrutil disable from Recovery mode, i.e. physical/console access to every machine. That doesn't scale for a rack fleet and makes remote reimaging impossible.
- SIP-on is the default after EACS and is fully remote-provisionable. We can Erase All Content & Settings → DEP re-enroll → MDM group assignment → bootstrap script → back in the prod pool, with zero datacenter hands. We just proved this end-to-end on a SIP-on M4 (clean EACS, came back into prod entirely via MDM + a one-shot bootstrap script).
- That gives us clean remote reimage and zero-touch return-to-prod for the whole fleet — a major operational win for hardware in a datacenter.

The only thing standing between us and making SIP-on the standard is this issue. Under SIP, the worker's system-level TCC permissions can no longer be granted by writing TCC.db directly (read-only under SIP) — they must come from an MDM PPPC profile, and a PPPC entry must anchor to the binary via a stable CodeRequirement. The worker binaries ship ad-hoc/unsigned, so there's nothing to anchor to.

Evidence from a freshly EACS'd SIP-on worker today (`macmini-m4-81`, `gecko-t-osx-1500-m4`, macOS 15.3):
```
$ codesign -dvvv /usr/local/bin/generic-worker-multiuser
Identifier=a.out
TeamIdentifier=not set        # ad-hoc — cannot be targeted by a PPPC CodeRequirement

$ sqlite3 .../TCC.db 'select service,client,auth_value from access where service="kTCCServiceScreenCapture"'
kTCCServiceScreenCapture|/usr/local/bin/generic-worker-multiuser|0    # DENIED
```

Our PPPC profile grants `FDA/Accessibility/AppleEvents` fine to Apple-signed path binaries (`sshd`, `osascript`, `bash`, `sqlite3`), but `ScreenCapture` for the worker can't be granted — no stable signing identity to reference. Net result: getDisplayMedia/display-capture tests fail on SIP-on workers. Example task Va2PAlD_TrG-jtXwuwLglA — the whole /screen-capture/getdisplaymedia.https.html set + permissions-policy-*display-capture* fail with:
NotAllowedError: The request is not allowed by the user agent or the platform in the current context.

These pass on the SIP-off fleet (system-DB grant), so today the choice is either keep SIP-off (no clean remote reimage, physical touch per host) or go SIP-on and lose the screen-capture suite. Signing the worker gets us both: zero-touch provisioning and full test parity.

The ask: ship the macOS generic-worker release artifacts (`generic-worker-multiuser`, `start-worker`) Developer ID–signed and notarized with a stable bundle identifier + TeamIdentifier (not ad-hoc). Then the PPPC profile can reference them durably, 

e.g.:
identifier "org.mozilla.generic-worker-multiuser" and anchor apple generic and certificate leaf[subject.OU] = <TeamID>
and grant kTCCServiceScreenCapture (and Camera/Microphone) under SIP. A stable identity also keeps the grants valid across worker-version bumps instead of going stale each release.

Bottom line: signing the worker binary is the last piece between Mozilla and fully zero-touch, remotely-reimageable macOS CI hardware at datacenter scale. Happy to validate a signed build on a SIP-on M4 and confirm the PPPC grant lands end-to-end.

[macmini-m4-81](https://firefox-ci-tc.services.mozilla.com/provisioners/releng-hardware/worker-types/gecko-t-osx-1500-m4/workers/mdc1/macmini-m4-81?sortBy=started&sortDirection=desc)

**Comment by hneiva** [comment]
I would like to add a few points to keep in mind here:

(1) The Taskcluster project runs on Community TC, not on Firefox TC instance; meaning if we want to add signing as part of CI, we will need a separate worker+pool+etc.

(2A) For this case specifically, I would recommend trying to use rcodesign on linux (via signingscript) instead of using a mac signer. This will likely require some work on Autograph.

(2B) If we decide to stick to mac signing (using iscript) - we will likely need another physical mac added to the fleet
